daffodil-core/src/main/scala/org/apache/daffodil/dsom/RuntimePropertyMixins.scala

```diff
@@ -158,6 +162,44 @@ trait TermRuntimeValuedPropertiesMixin
       Nope
   }
 
+  protected final lazy val choiceLengthExpr = {
```

Comment:
As you pointed out, alas, all this is unnecessary because it's not allowed to be an expression.

I'd suggest keeping this and making it a feature, but the amount of testing involved to really  support that, and lack of a compelling use case suggest you should just take this out.